core/state_transition.go

```diff
@@ -487,7 +488,7 @@ func (st *StateTransition) distributeTxFees() error {
 	caller := &vmcontext.SharedEVMRunner{EVM: st.evm}
```

Comment:
I would really like to get rid of this. The original purpose of the EVMRunner was to enforce that we couldn't use an evm more than once, so the existence of the SharedEVMRunner undermines the existence of EVMRunner and essentially renders it useless as a tool to enforce singular use of evms. So until we get rid of this I don't see that we get any benefit from the addition of the concept of EVMRunner.

I think it is also possible that the EVMRunner abstraction may not turn out to fit particularly well once we do the work to remove SharedEVMRunner and so the sooner we can do that work the better, because less will have been built on top of it and so less will need to be re-worked.

Reply:
piers.... 

1. If i use a normal EVMRunner here, that implies a hardfork
2. If i don't use, i need to do some custom thing for this particular case

On the next HF, i can then easily remove this, and use the normal VMRunner. Until then, sadly i can't.

So i ask you not to be so "absolute" about this... maybe wait a few months (next HF) until we're finally able to remove this, instead of saying that EVMRunner doesn't really do anything, because we have the SharedEVMRunner patch 😠